CICD/Services/ReadmeService.cs

```diff
@@ -0,0 +1,144 @@
+﻿// <copyright file="ReadmeService.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using CICDSystem.Exceptions;
+using CICDSystem.Guards;
+using CICDSystem.Services.Interfaces;
+
+// ReSharper disable InconsistentNaming
+namespace CICDSystem.Services;
+
+/// <inheritdoc/>
+internal sealed class ReadmeService : IReadmeService
+{
+    private const string MultiCommentStart = "<!--";
+    private const string MultiCommentStop = "-->";
+    private const string GitHubDirName = ".github";
+    private const string ReadMeFileName = "README.md";
+    private const string ProcessingCommentStart = "<!--PRE-PROCESSING-COMMENT-START-->"; // TODO: setup and str interpolation
+    private const string ProcessingCommentStop = "<!--PRE-PROCESSING-COMMENT-STOP-->";
+    private const string ProcessingUncommentStart = "<!--PRE-PROCESSING-UNCOMMENT-START-->";
+    private const string ProcessingUncommentStop = "<!--PRE-PROCESSING-UNCOMMENT-STOP-->";
+    private readonly IFile file;
+    private readonly string baseDirPath;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ReadmeService"/> class.
+    /// </summary>
+    /// <param name="directory">Manages directories.</param>
+    /// <param name="file">Manages files.</param>
+    /// <param name="path">Manages paths.</param>
+    /// <param name="findDirService">Searches for directories.</param>
+    public ReadmeService(
+        IDirectory directory,
+        IFile file,
+        IPath path,
+        IFindDirService findDirService)
+    {
+        EnsureThat.ParamIsNotNull(directory, nameof(directory));
+        EnsureThat.ParamIsNotNull(file, nameof(file));
+        EnsureThat.ParamIsNotNull(path, nameof(path));
+        EnsureThat.ParamIsNotNull(findDirService, nameof(findDirService));
+
+        this.file = file;
+
+        this.baseDirPath = findDirService.FindDescendentDir(directory.GetCurrentDirectory(), GitHubDirName)
+            .Replace('\\', '/');
+
+        this.baseDirPath = path.GetDirectoryName(this.baseDirPath)
+            .Replace('\\', '/');
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="FileNotFoundException">
+    ///     Thrown when the readme file is not found.
```

Comment:
Refactor readme to README
```suggestion
    ///     Thrown when the README file is not found.
```